### PR TITLE
Fix bug

### PR DESCRIPTION
**Pull request by lth08091998**
Profile API response

screenshot from 2018-11-18 14-36-40

### CODE REVIEW

File: kidsbook/user/views.py [hunk 1]
```diff
@@ -305,14 +305,24 @@ def list(self, request, **kargs):
...
+                comments_data = CommentSerializer(comments, many=True).data
+                    
                 response_data['num_comment'] = len(comments)
-
+                print(comments_data)
                 post_like_received = 0
                 if('user_posts' in response_data):
                     response_data['user_posts'] = list(map(lambda post: post['id'], response_data['user_posts']))
                     for post_id in response_data['user_posts']:
                         post_like = UserLikePost.objects.all().filter(post=Post.objects.get(id=post_id)).filter(like_or_dislike=True)
```
Comment (snguyenthanh) [inline kidsbook/user/views.py]: No need for `.all` as `UserLikePost.objects.filter` is sufficient

File: kidsbook/user/views.py [hunk 2]
```diff
@@ -305,14 +305,24 @@ def list(self, request, **kargs):
...
                         post_like_received += len(post_like)
+                for idx, post_id in enumerate(response_data['user_posts']):
+                    post_comments = list(filter(lambda comment: str(comment['post']) == str(post_id), comments_data))
+                    for idx2, current_comment in enumerate(post_comments):
+                        post_comments[idx2] = current_comment['id']
+                    post_likes_received = len(UserLikePost.objects.all().filter(post=Post.objects.get(id=post_id)).filter(like_or_dislike=True))
```
Comment (snguyenthanh) [inline kidsbook/user/views.py]: No need for `.all` as `UserLikePost.objects.filter` is sufficient

File: kidsbook/user/views.py [hunk 3]
```diff
@@ -305,14 +305,24 @@ def list(self, request, **kargs):
                 if('role' in response_data):
                     response_data['role'] = response_data['role']['id']
                 comments = Comment.objects.all().filter(creator=user)
+                comments_data = CommentSerializer(comments, many=True).data
+                    
                 response_data['num_comment'] = len(comments)
-
+                print(comments_data)
```
Comment (snguyenthanh) [inline kidsbook/user/views.py]: Remove this debugging print()

File: kidsbook/user/views.py [hunk 4]
```diff
@@ -321,8 +331,8 @@ def list(self, request, **kargs):
                 if(len(time_arr) > 0):
                     response_data['time_history'] = time_arr
                 return Response({'data': response_data})
-        except Exception:
-            pass
+        except Exception as e:
+            print(e)
```
Comment (snguyenthanh) [inline kidsbook/user/views.py]: Can you return `e` as a `string` error message with status code 400 ?

**Comment by Ang-YC** [comment]
I keep getting the following error, any idea?

```
Internal Server Error: /api/v1/group/f2b62cd8-7a45-4330-b2ec-e3b54d24554c/posts/
Traceback (most recent call last):
  File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/django/core/handlers/base.py", line 126, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/django/core/handlers/base.py", line 124, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/rest_framework/views.py", line 483, in dispatch
    response = self.handle_exception(exc)
  File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/rest_framework/views.py", line 443, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/rest_framework/views.py", line 480, in dispatch
    response = handler(request, *args, **kwargs)
  File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/rest_framework/generics.py", line 241, in get
    return self.list(request, *args, **kwargs)
  File "/Users/ang/NUS/CS3216/kidsbook-backend/kidsbook/post/views.py", line 56, in list
    response_data = serializer.data
  File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/rest_framework/serializers.py", line 765, in data
    ret = super(ListSerializer, self).data
  File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/rest_framework/serializers.py", line 262, in data
    self._data = self.to_representation(self.instance)
  File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/rest_framework/serializers.py", line 683, in to_representation
    self.child.to_representation(item) for item in iterable
  File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/rest_framework/serializers.py", line 683, in <listcomp>
    self.child.to_representation(item) for item in iterable
  File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/rest_framework/serializers.py", line 527, in to_representation
    ret[field.field_name] = field.to_representation(attribute)
  File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/rest_framework/fields.py", line 1855, in to_representation
    return method(value)
  File "/Users/ang/NUS/CS3216/kidsbook-backend/kidsbook/serializers.py", line 90, in get_filtered_content
    return censor(obj.content)
  File "/Users/ang/NUS/CS3216/kidsbook-backend/kidsbook/utils.py", line 51, in censor
    return profanity.censor(text)
  File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/profanity/profanity.py", line 66, in censor
    cen = "".join(get_censor_char() for i in list(word))
  File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/profanity/profanity.py", line 66, in <genexpr>
    cen = "".join(get_censor_char() for i in list(word))
  File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/profanity/profanity.py", line 37, in get_censor_char
    return _censor_pool.pop(random.randrange(len(_censor_pool)))
IndexError: pop from empty list
```

**Comment by snguyenthanh** [comment]
> I keep getting the following error, any idea?
> 
> ```
> Internal Server Error: /api/v1/group/f2b62cd8-7a45-4330-b2ec-e3b54d24554c/posts/
> Traceback (most recent call last):
>   File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/django/core/handlers/exception.py", line 34, in inner
>     response = get_response(request)
>   File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/django/core/handlers/base.py", line 126, in _get_response
>     response = self.process_exception_by_middleware(e, request)
>   File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/django/core/handlers/base.py", line 124, in _get_response
>     response = wrapped_callback(request, *callback_args, **callback_kwargs)
>   File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
>     return view_func(*args, **kwargs)
>   File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/django/views/generic/base.py", line 68, in view
>     return self.dispatch(request, *args, **kwargs)
>   File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/rest_framework/views.py", line 483, in dispatch
>     response = self.handle_exception(exc)
>   File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/rest_framework/views.py", line 443, in handle_exception
>     self.raise_uncaught_exception(exc)
>   File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/rest_framework/views.py", line 480, in dispatch
>     response = handler(request, *args, **kwargs)
>   File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/rest_framework/generics.py", line 241, in get
>     return self.list(request, *args, **kwargs)
>   File "/Users/ang/NUS/CS3216/kidsbook-backend/kidsbook/post/views.py", line 56, in list
>     response_data = serializer.data
>   File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/rest_framework/serializers.py", line 765, in data
>     ret = super(ListSerializer, self).data
>   File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/rest_framework/serializers.py", line 262, in data
>     self._data = self.to_representation(self.instance)
>   File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/rest_framework/serializers.py", line 683, in to_representation
>     self.child.to_representation(item) for item in iterable
>   File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/rest_framework/serializers.py", line 683, in <listcomp>
>     self.child.to_representation(item) for item in iterable
>   File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/rest_framework/serializers.py", line 527, in to_representation
>     ret[field.field_name] = field.to_representation(attribute)
>   File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/rest_framework/fields.py", line 1855, in to_representation
>     return method(value)
>   File "/Users/ang/NUS/CS3216/kidsbook-backend/kidsbook/serializers.py", line 90, in get_filtered_content
>     return censor(obj.content)
>   File "/Users/ang/NUS/CS3216/kidsbook-backend/kidsbook/utils.py", line 51, in censor
>     return profanity.censor(text)
>   File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/profanity/profanity.py", line 66, in censor
>     cen = "".join(get_censor_char() for i in list(word))
>   File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/profanity/profanity.py", line 66, in <genexpr>
>     cen = "".join(get_censor_char() for i in list(word))
>   File "/Users/ang/.pyenv/versions/3.5.2/lib/python3.5/site-packages/profanity/profanity.py", line 37, in get_censor_char
>     return _censor_pool.pop(random.randrange(len(_censor_pool)))
> IndexError: pop from empty list
> ```

This is an error caused by an external library [profanity](https://github.com/ben174/profanity).

I read the source code but still unsure what caused the error.

**Pull request merged**